### PR TITLE
Added tutorial doc on how to add a Space to Arxiv

### PR DESCRIPTION
**Pull request by abidlabs**
Adds a step-by-steps post on how to add a Space to ArXiv. Closes: #526

### CODE REVIEW

File: docs/hub/arxiv.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    For example, the [docformer_for_document_classification](https://huggingface.co/spaces/iakarshu/docformer_for_document_classification) Space loads the LayoutLM [like this](https://huggingface.co/spaces/iakarshu/docformer_for_document_classification/blob/main/modeling.py#L484) and include the string `"microsoft/layoutlm-base-uncased"`:
+
+    ```py
+    from transformers import LayoutLMForTokenClassification
+    
+    layoutlm_dummy = LayoutLMForTokenClassification.from_pretrained("microsoft/layoutlm-base-uncased", num_labels=1)
+    ```
+
+    More instructions on [building a demo on Hugging Face Spaces are here](./spaces-overview).
```
Comment (osanseviero) [inline docs/hub/arxiv.md:36]: Maybe you can add something that says that you can use `huggingface_hub` to download a model programmatically. For most cases they won't have out of the box integration with a library

File: docs/hub/arxiv.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    from transformers import LayoutLMForTokenClassification
+    
+    layoutlm_dummy = LayoutLMForTokenClassification.from_pretrained("microsoft/layoutlm-base-uncased", num_labels=1)
+    ```
+
+    More instructions on [building a demo on Hugging Face Spaces are here](./spaces-overview). 
+
+4. That's it! As soon as your Space is built, Hugging Face will detect that it is associated with the model and your Space should appear in the Demo tab on ArXiv in a few minutes.
+
+    *Note*: after your Space is built, you can verify that it is linked to your model. A "Linked Models" button should appear in the top right corner of the Space and link to the model.
```
Comment (osanseviero) [inline docs/hub/arxiv.md:40]: Add a screenshot here

**Review by osanseviero — COMMENTED** [review]
Very cool! I left some comments

File: docs/hub/_toctree.yml [hunk 3]
```diff
@@ -135,7 +135,9 @@
     - local: spaces-more-ways-to-create
       title: More ways to create Spaces
     - local: spaces-github-actions
-      title: Managing Spaces with Github Actions      
+      title: Managing Spaces with Github Actions 
+    - local: arxiv
```
Comment (julien-c) [inline docs/hub/_toctree.yml:139]:
```suggestion
    - local: spaces-add-to-arxiv
```

File: docs/hub/arxiv.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# How to Add a Space to ArXiv
+
+Demos on Hugging Face Spaces allow a wide audience to try out state-of-the-art machine
+learning research without writing any code. Hugging Face and ArXiv have collaborated 
+to embed these demos directly along side papers on ArXiv!
+
+Thanks to this integration, users can now find the most popular demos for a paper on its arXiv abstract page. For example, if you want to try out demos of the LayoutLM document classification model, you can go to [LayoutLM's arXiv page](https://arxiv.org/abs/1912.13318), and navigate to the demo tab. You will see open-source demos built by the machine learning community for this model.
```
Comment (julien-c) [inline docs/hub/arxiv.md:7]:
```suggestion
Thanks to this integration, users can now find the most popular demos for a paper on its arXiv abstract page. For example, if you want to try out demos of the LayoutLM document classification model, you can go to [the LayoutLM paper's arXiv page](https://arxiv.org/abs/1912.13318), and navigate to the demo tab. You will see open-source demos built by the machine learning community for this model.
```

File: docs/hub/arxiv.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+We'll cover the steps on how to add your Space to ArXiv and have it show up in  the Demos tab. There are a few prerequisites for this:
+
+**Prerequisites**
+
+* There's an existing paper on ArXiv that you'd like to create a demo for
+* The paper is **associated with a model** that is on the Hugging Face Model Hub (or can be uploaded there)
```
Comment (julien-c) [inline docs/hub/arxiv.md:15]:
```suggestion
* The paper is **associated with a model or a dataset** that is on the Hugging Face Model Hub (or can be uploaded there)
```

File: docs/hub/arxiv.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+* You have built or (can build) a demo for the model on Spaces
```
Comment (julien-c) [inline docs/hub/arxiv.md:16]:
```suggestion
* You have built or (can build) a demo for the model or the dataset on Spaces
```

File: docs/hub/arxiv.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+**Steps to Add Your Space**
+
+1. First, upload the model associated with the ArXiv paper onto the Hugging Face Model Hub if it is not already there. ([Detailed instructions are here](./models-uploading))
```
Comment (julien-c) [inline docs/hub/arxiv.md:20]:
```suggestion
1. First, upload the model or dataset associated with the ArXiv paper onto the Hugging Face Hub if it is not already there. ([Detailed instructions are here](./models-uploading))
```

File: docs/hub/arxiv.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+2. When writing the model card (README.md) for the model, include a link to the ArXiv paper. It's good practice to include a full citation as well. You can see an example of a link and a citation on the [LayoutLM model card](https://huggingface.co/microsoft/layoutlm-base-uncased)
+
+    *Note*: you can verify this step has been carried out successfully by seeing if an ArXiv button appears above the model card. In the case of LayoutLM, the button says: "arxiv:1912.13318" and links to the LayoutLM paper on ArXiv.
+
+3. Then, create a demo on Spaces that loads this model. Somewhere within the code, the full path to the model must be included in order for Hugging Face to detect that a Space is associated with a particular model.
```
Comment (julien-c) [inline docs/hub/arxiv.md:26]:
```suggestion
3. Then, create a demo on Spaces that loads this model or dataset. Somewhere within the code, the model or dataset Hub id must be included in order for Hugging Face to detect that a Space is associated with it.
```

File: docs/hub/arxiv.md (hunk 1 above)
Comment (julien-c) [reply docs/hub/arxiv.md:36]: you can also link to the page where we describe that if automatic model/dataset detection does not work you can specify them explicitly in the README metadata

**Review by julien-c — COMMENTED** [review]
My two feedbacks:
- inter-link more to other doc pages, e.g. the one on Space README.md metadata
- this is not just for models but also for datasets. As we try to push for more recognition for datasets (including in published papers) would be cool to highlight both

**Comment by abidlabs** [comment]
Thanks for the great suggestions @osanseviero and @julien-c! 

* Updated the URL to `spaces-add-to-arxiv`
* Added references to datasets
* Added more inter-linking
* Added screenshots

**Review by osanseviero — APPROVED** [review]
Looks nice in https://moon-ci-docs.huggingface.co/docs/hub/pr_530/en/spaces-add-to-arxiv :fire:

**Comment by osanseviero** [comment]
Feel free to merge if you're happy with it :)

**Comment by abidlabs** [comment]
Thanks @osanseviero!

**Review by julien-c — COMMENTED** [review]
yay! quite cool

**Comment by julien-c** [comment]
maybe we could have added a screenshot of what it looks like on Arxiv and/or a link to the blogpost where we announced the integration

**Comment by abidlabs** [comment]
Sure thanks for the suggestion @julien-c, will add later this week

**Pull request merged**